src/features/profile/pages/ChangeEmail/ChangeEmail.native.test.tsx

```diff
@@ -36,41 +36,46 @@ server.use(
 describe('<ChangeEmail/>', () => {
   beforeEach(simulateUpdateEmailSuccess)
 
-  it('should render correctly', () => {
-    const renderAPI = renderChangeEmail()
-    expect(renderAPI).toMatchSnapshot()
+  it('should render correctly', async () => {
+    renderChangeEmail()
+
+    await waitFor(() => {
+      expect(screen).toMatchSnapshot()
+    })
   })
 
   describe('email change already in progress', () => {
+    const alreadyInProgressText =
+      'Une demande a été envoyée à ta nouvelle adresse. Tu as 24h pour la valider. Pense à vérifier tes spams.'
+
     it('should render correctly', async () => {
       simulateCurrentEmailChange()
-      const renderAPI = renderChangeEmail()
+      renderChangeEmail()
+      await screen.findByText(alreadyInProgressText)
 
-      await superFlushWithAct()
-      expect(renderAPI).toMatchSnapshot()
+      expect(screen).toMatchSnapshot()
```

Comment:
Le findBy sert de waitFor ici non ? Le waitFor ne suffisait pas ?

Reply:
nous avons ajouté plein de findBy pour attendre que tout les re-render soient fini avant de faire l'assertion

waitFor ré-éxécute la fonction en boucle jusqu'à tant que ça passe ou qu'il y ai un timeout

faire un waitFor sur un snapshot ou sur la vérification d'un `not.toHaveBeenCalled()` me parait non pertinent car il risque de passer dans les premier run alors que si on attends suffisamment que tout les re-render soient fini, ça ne devrait pas passer